polynote-server/src/main/scala/polynote/server/KernelSubscriber.scala

```diff
@@ -34,7 +34,11 @@ class KernelSubscriber private[server] (
   }
 
   def close(): UIO[Unit] = closed.succeed(()).unit *> process.interrupt.unit
-  def update(update: NotebookUpdate): Task[Unit] = publisher.update(id, update) *> ZIO(lastLocalVersion.set(update.localVersion)) *> ZIO(lastGlobalVersion.set(update.globalVersion))
+
+  def update(update: NotebookUpdate): Task[Unit] =
+    ZIO.effectTotal(lastLocalVersion.set(update.localVersion)) *>
```

Comment:
Is there any difference between `effectTotal` and `success`?

Reply:
I think `ZIO.succeed` is an alias for `ZIO.effectTotal`... but I think of `succeed` as taking a pure value (even though it is suspended in reality) and `effectTotal` as suspending the argument.

Reply:
I see, interesting. I guess the idea is that it's an analogue for (in cats-effect) `IO.pure(function())` vs `IO(function())`, where in the former `function()` is called and its result is suspended in the `IO` but in the latter `function()` itself is suspended and not called until the `IO` is evaluated?

But in this case it's more for readability/intent?